BuildingRefrigerator/BuildingRefrigerator/Program.cs

```diff
@@ -0,0 +1,175 @@
+﻿using BuildingRefrigerator;
+class Program
+{
+    static void Main()
+    {
+        Options choice;
+        Guid idItem;
+        try
+        {
+            refrigerator myRefrigerator = BuildingRefrigerator();
+            printMenu();
+            if (!Options.TryParse(Console.ReadLine(), out choice)) throw new Exception("invalid choice");
+            while ((int)choice != 100)
+            {
+                switch (choice)
+                {
+                    case Options.PrintAll:
+                        Console.WriteLine("print all of the items in the refrigerator:\n{0}", myRefrigerator);
+                        break;
+                    case Options.FreeSpaceInTheFridge:
+                        Console.WriteLine("Prints the free space in the refrigerator:{0}\n", myRefrigerator.HowMuchSpaceIsLeftOnTheRefrigerator());
+                        break;
+                    case Options.InsertItem:
+                        Console.WriteLine("You put an item in the fridge:");
+                        myRefrigerator.InsertItem(buildingItem());
+                        break;
+                    case Options.TakingItemOut:
+                        Console.WriteLine("Insert an item ID to remove it from the water:");
+                        if (!Guid.TryParse(Console.ReadLine(), out idItem)) throw new Exception("idItem is invalid");
+                        myRefrigerator.TakingAnItemOut(idItem);
+                        break;
+                    case Options.CleaningRefrigerator:
+                        Console.WriteLine("we are cleaning the refrigerator ");
+                        myRefrigerator.CleaningRefrigerator();
+                        break;
+                    case Options.WhatDoYouWantToEat:
+                        whatDoYouWantToEat(myRefrigerator);
+                        break;
+                    case Options.PrintingProductsAccordingToExpirationDate:
+                        printItems(myRefrigerator.SortedByExpirationDate());
+                        break;
+                    case Options.PrintingShelvesAccordingToAvailableSpace:
+                        printShelves(myRefrigerator.SortByAvailableShelfSpace());
+                        break;
+                    case Options.PrintingRefrigeratorsAccordingToAvailableSpace:
+                        sortRefrigerators();
+                        break;
+                    case Options.PreparingRefrigeratorForShopping:
+                        Console.WriteLine("We are preparing refrigerator for shopping");
+                        myRefrigerator.Shopping();
+                        break;
+                    default:
+                        break;
+                }
+
+                printMenu();
+                if (!Options.TryParse(Console.ReadLine(), out choice)) throw new Exception("invalid choice");
+            }
+        }
+        catch (Exception ex) { Console.WriteLine(@"Dear Customer: 
+{0}", ex.Message); }
```

Comment:
Bad UX here to exit the program for any user mistake.

Reply:
In fact, it is better to turn the IF into a WHILE until it receives a correct value, in the event of a wrong user input?